src/Mvc/Mvc.Core/src/ApplicationModels/ApiBehaviorApplicationModelProvider.cs

```diff
@@ -52,6 +52,8 @@ internal sealed class ApiBehaviorApplicationModelProvider : IApplicationModelPro
                 new InferParameterBindingInfoConvention(modelMetadataProvider, serviceProviderIsService);
             ActionModelConventions.Add(convention);
         }
+
+        ActionModelConventions.Add(new HttpResultMetadataConvention());
```

Comment:
Nit: Since this isn't gated on anything, let's just add it when we construct `ActionModelConventions` along with `new ApiVisibilityConvention()`.